### DISCUSSION TITLE
APISIX Ingress Controller + Gateway as an option in the framework

### DISCUSSION THREAD

**Discussion by amitbansal26** [Sunbird-Ed/Community#541] [General Discussion]
Can APISIX Ingress Controller be considered instead of nginx + KONG. This can act  both as ingress and API Gateway Additionally it has plugin to connect to keycloak  and countless other plugins. This is also a part of CNCF and is fully open source as compared to kong with Dual Licensing . Becomes more viable as a DPG.

Since I am posting for first time please feel free to point me to any such discussion if already there .....

**Comment by Krishnaj20** [comment]
@vinukumar-vs @gandham-santhosh - Please respond to this thread.

**Reply by chitranshu-keshav** [reply]
@vinukumar-vs @gandham-santhosh Can you please get back on this

**Comment by gandham-santhosh** [comment]
Hi, @amitbansal26 thank you for sharing this recommendation. Given APISIX is also part of Cloud native landscape list we can definitely explore but it still has not sandboxed/graduated. 
This is a good start to discuss about open-source API Managers :) 
What do you think about https://github.com/emissary-ingress/emissary which is envoy based developed using python and go rather nginx lua? This might give good throughput I guess  
Also, can you please share if you have come across any organizations/applications using APISIX on a large scale?

**Reply by amitbansal26** [reply]
@gandham-santhosh  Yes Agreed but  emissary ingress does not support JWT, OAUTH2 and openid ingress out of the box ... here is the detailed comparison matrix for reference also performance of apisix is way ahead of emissary 

https://api7.ai/blog/apisix-ingress-vs-emissary-ingress

I may not have used apisix but apisix looks more promising . In the enterprise i have mostly used paid gateways like APIGEE as per the client requirements
